src/pygeoprocessing/routing/routing.pyx

```diff
@@ -4159,6 +4216,131 @@ def calculate_subwatershed_boundary(
         '(calculate_subwatershed_boundary): watershed building 100% complete')
 
 
+def detect_lowest_drain_and_sink(dem_raster_path_band):
+    """Find the lowest drain and sink pixel in the DEM.
+
+    This function is used to specify conditions to DEMs that are known to
+    have one real sink/drain, but may have several numerical sink/drains by
+    detecting both the lowest pixel that could drain the raster on an edge
+    and the lowest internal pixel that might sink the whole raster.
+
+    Example:
+        raster A contains the following
+            * pixel at (3, 4) at 10m draining to a nodata  pixel
+            * pixel at (15, 19) at 11m draining to a nodata pixel
+            * pixel at (19, 21) at 10m draining to a nodata pixel
+            * pit pixel at (10, 15) at 5m surrounded by non-draining pixels
+            * pit pixel at (25, 15) at 15m surrounded by non-draining pixels
+            * pit pixel at (2, 125) at 5m surrounded by non-draining pixels
+
+        The result is two pixels indicating the first lowest edge and first
+        lowest sink seen:
+            drain_pixel = (3, 4), 10
+            sink_pixel = (10, 15), 5
+
+    Args:
+        dem_raster_path_band (tuple): a raster/path band tuple to detect
+            sinks in.
+
+    Return:
+        (drain_pixel, drain_height, sink_pixel, sink_height) -
+            two (x, y) tuples with corresponding heights, first
+            list is for edge drains, the second is for pit sinks. The x/y
+            coordinate is in raster coordinate space and `height` is the

```

Comment:
Should `height` be double ticks here?

Reply:
It wasn't referring to an actual parameter which is why i didn't have double ticks, just the suffix of those but I changed it in 0fd7e78 from `height` to _height so that's more obvious.